src/components/GoodsPage/GoodsPage.tsx

```diff
@@ -0,0 +1,173 @@
+import React, {  useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { Table, Image, TableProps, Divider } from "antd";
+
+import { AppDispatch } from "../../store/store";
+import { actions, GoodSelectors } from "store/goodsSlice";
+import { actionsCategories } from "store/categoriesSlice";
+import { Link } from "react-router-dom";
+import { selectorsCategories } from "store/categoriesSlice";
+import { ColumnsType } from "antd/es/table";
+import { Good, GoodsSearch } from "../../api/api";
+import Search from "antd/es/input/Search";
+
+import { Loader } from "../Loader";
+import { ErrorToast } from "../ErrorToast";
+
+export const GoodsPage: React.FC = () => {
+  const dispatch = useDispatch<AppDispatch>();
+
+  const goods = useSelector(GoodSelectors.getMapGoods);
+  const loading = useSelector(GoodSelectors.getIsLoadingSelector);
+  const error = useSelector(GoodSelectors.getIsErrorSelector);
+  const { categories } = useSelector(selectorsCategories.getCategories);
+  const [value, setValue] = useState("");
+  const [params, setParams] = useState<GoodsSearch>({});
+
+  useEffect(() => {
+    dispatch(actionsCategories.actions.fetchCategories());
+  }, []);
+
+  useEffect(() => {
+    dispatch(actions.fetchGoods(params));
+  }, [params]);
+
+  const onSearch = () => {
+    setParams((prevState) => {
+      return { ...prevState, text: value };
+    });
+    setValue("");
+    console.log(params.text);
+  };
+  const onChange: TableProps<Good>["onChange"] = (
+    pagination,
+    filters,
+    sorter: { [key: string]: any }
+  ) => {
+    setParams((prevState) => {
+      let newState = prevState;
```

Comment:
ты мутируешь стейт, т.к. передал объект по ссылке, надо новую копию делать 
```suggestion
      let newState = { ...prevState };
```